### PR TITLE
Added an optional prefix for generated types

### PR DESCRIPTION
**Pull request by pirrmann**

### CODE REVIEW

File: src/RegexProvider/RegexProvider.fs [hunk 1]
```diff
@@ -11,10 +11,20 @@ module internal TypedRegex =
     let typedRegex() = 
         let regexType = erasedType<Regex> thisAssembly rootNamespace "Regex"
         regexType.DefineStaticParameters(
-            parameters = [ProvidedStaticParameter("pattern", typeof<string>)], 
+            parameters =
+                [
+                    ProvidedStaticParameter("pattern", typeof<string>)
+                    ProvidedStaticParameter("generatedMethodsPrefix", typeof<string>, "")
```
Comment (vasily-kirichenko) [inline src/RegexProvider/RegexProvider.fs]: I don't quite like that user can provide any prefix. He will have to provide the same string again and again, for each type. Maybe it'd be better to add `Typed` prefix by default and no prefix if a boolean static parameter provided, like this:
```fsharp
// "TypedXXX"
type WordRegex = Regex< @"(?<Word>\w+)">
// The old behavior, for backward compatibility
type WordRegex = Regex< @"(?<Word>\w+)", NoMethodPrefix = true>
```

Comment (pirrmann) [inline src/RegexProvider/RegexProvider.fs]: If you don't mind breaking compatibility by default, of course this is much more convenient

**Comment by vasily-kirichenko** [comment]
Looks great. Thanks!

**Comment by vasily-kirichenko** [comment]
I have no rights to publish NuGet package though.

**Comment by dtchepak** [comment]
@vasily-kirichenko: I'll push the nuget package now. If you give me your nuget username I'll add you as package owner.

**Comment by vasily-kirichenko** [comment]
My NuGet user name is vaskir.

**Comment by dtchepak** [comment]
Just confirming before I push this -- this is a breaking change? If so will need to update the docs accordingly.

**Comment by dtchepak** [comment]
@vasily-kirichenko: you've got nuget access now. I added a page with the [release procedure](https://github.com/fsprojects/FSharp.Text.RegexProvider/wiki).

**Comment by vasily-kirichenko** [comment]
Yes. I think we can make it version 1.0.0 as well.

**Comment by pirrmann** [comment]
Yes, that became a breaking change along the way. I can update the docs in my PR if you want

**Comment by dtchepak** [comment]
@pirrmann: that would be great, thanks!

**Comment by vasily-kirichenko** [comment]
As the PR has been merged, open a new one, please.

**Pull request merged**